components/builder-api/src/server/resources/origins.rs

```diff
@@ -575,8 +576,12 @@ fn create_origin_secret(
     }
 
     // get metadata from secret payload
-    let secret_metadata = match BoxKeyPair::secret_metadata(body.value.as_bytes()) {
-        Ok(res) => res,
+    let ciphertext = WrappedSealedBox::from(body.value.as_str());
+    let secret_metadata = match BoxKeyPair::secret_metadata(&ciphertext) {
+        Ok(res) => {
+            debug!("Secret Metadata: {:?}", res);
```

Comment:
Is it ok to leave this `debug!` in?  Just wondering about sensitive info leaking into the logs.

Reply:
I think so, because if you look at [`secret_metadata`](https://github.com/habitat-sh/core/blob/master/components/core/src/crypto/keys/box_key_pair.rs#L332), you'll see it's returning is a `BoxSecret`:

```
pub struct BoxSecret<'a> {
    pub sender: &'a str,
    pub ciphertext: Vec<u8>,
    pub receiver: Option<&'a str>,
    pub nonce: Option<Nonce>,
}
```

The key here is that the key field is *cipher*text, not plaintext. In a functional crypto system, cipher text should be indistinguishable from random bytes. The only way to recover the plaintext would be to have the appropriate keys, which this struct lacks. It does contain the `nonce`, which I think would possibly be better to *not* share, but in the context of internal debug logs, I think its value in debugging outweighs the risk in exposing it. Then again, I'm not a security expert. In any case, this is not a net change, so perhaps we should file a separate issue to consider a non-derived implementation of `Debug` for `BoxSecret` which would expose fewer details.

See also https://en.wikipedia.org/wiki/Ciphertext-only_attack